src/Microsoft.ML.HalLearners/SymSgdClassificationTrainer.cs

```diff
@@ -638,10 +643,10 @@ private TPredictor TrainCore(IChannel ch, RoleMappedData data, LinearModelParame
             int numFeatures = data.Schema.Feature.Value.Type.GetVectorSize();
             var cursorFactory = new FloatLabelCursor.Factory(data, CursOpt.Label | CursOpt.Features | CursOpt.Weight);
```

Comment:
>| CursOpt.Weight [](start = 100, length = 16)

You don't accept weight in options, and I don't see any calls to Weight column.
Would be nice to remove it from cursor creation.